lambdas/Dockerfile

```diff
@@ -1,10 +1,11 @@
 FROM public.ecr.aws/lambda/nodejs:14
 
-# Copy script files and package.json to lambda root
-COPY *.js package.json ${LAMBDA_TASK_ROOT}
+################################
+# This dockerfile depends on npm ci and typescipt having already been compiled
```

Comment:
This needs some discussion, probably.

It is a bit of a deviation from "normal" docker build best practices where the dockerfile can build the project from scratch without requiring prior steps. But it is the only reasonable way I could figure out (with @GPaoloni's help) to use the external repo and maintain a reasonable build context for the docker build command. This change would also need to be made in the HRM Dockerfile for it to be able to use the local repo.

I think the Right™ way is probably to publish to NPM, but that adds more complexity than we are ready for.

Totally open to thoughts and suggestions

Reply:
I really don't like the idea of moving local modules out to separate NPM package. Doing that to make the Dockerfile easier or 'more correct' is definitely a case of the tail wagging the dog, moving us further away from good practices about how we structure our repos and giving us unnecessary friction and maintenance headaches.

Besides, would having the local modules in the dockers build context be unreasonable? We copy the project we are building and a couple of sibling projects into the container's directory structure, build the project in its subdirectory then move the finished build up into the root - doesn't seem so bad to me but I could be missing something. 

But creating the image from the built output seems fine too.

Reply:
Sorry. That code snippet/comment is outdated. After some discussion with Gian we moved to the dockerfile with full repo context that can do a full build from scratch. See the updated docker file [here](https://github.com/techmatters/hrm/pull/240/files#diff-47430fe8f45fea5be626def7d794bcc87aac1ac79dddb9a7e9e2f3b75db7fdf1R1).

This also means we'll need to modify the hrm-services docker build similarly which is something to keep in mind.

Reply:
Sorry, I saw the code was outdated but thought the discussion was still relevant, should have reviewed the actual updated versions more carefully :facepalm:

Reply:
No worries. I should have updated the convo thread!